src/main/java/com/zaxxer/hikari/HikariConfig.java

```diff
@@ -766,26 +766,18 @@ public void validate()
          throw new IllegalArgumentException("poolName cannot contain ':' when used with JMX");
       }
 
-      if (driverClassName != null && jdbcUrl == null) {
-         LOGGER.error("jdbcUrl is required with driverClassName");
-         throw new IllegalArgumentException("jdbcUrl is required with driverClassName");
+      if (dataSource != null) {
+         LOGGER.debug("{} - using dataSource={}", poolName, dataSource);
       }
-      else if (driverClassName != null && dataSourceClassName != null) {
-         LOGGER.error("cannot use driverClassName and dataSourceClassName together");
-         throw new IllegalArgumentException("cannot use driverClassName and dataSourceClassName together");
-      }
-      else if (jdbcUrl != null && dataSourceClassName != null) {
-         LOGGER.warn("using dataSourceClassName and ignoring jdbcUrl");
```

Comment:
those 'ignoring' message or exception is 'our' creation. specifying both would not break anything as pool proceeds with initiating data source in order of dataSource then dataSourceClassName then jdbcUrl
However,  ...to avoid tickets ..adding these checks and exceptions ..in few mins